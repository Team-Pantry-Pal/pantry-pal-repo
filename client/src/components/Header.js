import React, { Component, Fragment } from 'react';
import { Collapse, Container, Nav, Navbar, NavbarBrand, NavbarToggler } from 'reactstrap';
import SignInNav from './SignInNav';
import AppNav from './AppNav';

class Header extends Component {
  state = {
    isOpen: false
  };
  toggle = () => {
    this.setState((prevState) => ({ isOpen: !prevState.isOpen }));
  };
  render() {
    return (
      <Fragment>
        <Navbar dark color="dark" expand="sm">
          <Container fluid>
            <NavbarBrand href="/">PantryPal</NavbarBrand>
            <NavbarToggler onClick={this.toggle} />
            <Collapse navbar isOpen={this.state.isOpen}>
              <Nav className="ml-auto" navbar>
                {
                  this.props.isLoggedIn ? (
                    <AppNav
                      user={this.props.user}
                      logOutUser={this.props.logOutUser}
                    />
                  ) : (
                    <SignInNav
                      push={this.props.push}
                      logInUser={this.props.logInUser}
                    />
                  )
                }
              </Nav>
            </Collapse>
          </Container>
        </Navbar>
      </Fragment>
    );
  }
}

export default Header;