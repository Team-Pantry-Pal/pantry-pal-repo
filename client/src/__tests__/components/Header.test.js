import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Header isLoggedIn={false} />);
});

test('Should render Header correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Should handle toggle', () => {
  wrapper.find('NavbarToggler').simulate('click');
  expect(wrapper.state('isOpen')).toBe(true);
});

test('Should render Header with AppNav content', () => {
  wrapper.setProps({ isLoggedIn: true });
  expect(wrapper).toMatchSnapshot();
});