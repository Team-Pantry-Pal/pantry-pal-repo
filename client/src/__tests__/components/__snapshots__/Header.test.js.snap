// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<Fragment>
  <Navbar
    color="dark"
    dark={true}
    expand="sm"
    tag="nav"
  >
    <Container
      fluid={true}
      tag="div"
    >
      <NavbarBrand
        href="/"
        tag="a"
      >
        PantryPal
      </NavbarBrand>
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <SignInNav />
        </Nav>
      </Collapse>
    </Container>
  </Navbar>
</Fragment>
`;

exports[`Should render Header with AppNav content 1`] = `
<Fragment>
  <Navbar
    color="dark"
    dark={true}
    expand="sm"
    tag="nav"
  >
    <Container
      fluid={true}
      tag="div"
    >
      <NavbarBrand
        href="/"
        tag="a"
      >
        PantryPal
      </NavbarBrand>
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <AppNav />
        </Nav>
      </Collapse>
    </Container>
  </Navbar>
</Fragment>
`;
